### PR TITLE
Create and run local network

### PR DESCRIPTION
**Pull request by madsbk**
This PR implements `utils.run_on_local_network()`, which spawns, connects, and runs a number of workers.

```python

def run_on_local_network(
    n_workers, worker_func, worker_args=None, server_address=None, ucx_options_list=None
):
    """
    Creates a local UCX network of `n_workers` that runs `worker_func`

    Parameters
    ----------
    n_workers : int
        Number of workers (nodes) in the network.
    worker_func: callable (can be a coroutine)
        Function that each worker execute. 
        Must have signature: `worker(rank, eps, args)` where
            - rank is the worker id
            - eps is a dict of ranks to ucx endpoints
            - args given here as `worker_args`
    worker_args: object
        The argument to pass to `worker_func`.
    server_address: str
        Server address for the workers. If None, get_address() is used.
    ucx_options_list: list of dict
        Options to pass to UCX when initializing workers, one for each worker.

    Returns
    -------
    dev_names : str
        Names of the closest net devices
    """
```

### CODE REVIEW

File: ucp/utils.py [hunk 1]
```diff
@@ -39,3 +45,133 @@ def get_address(ifname=None):
             s.fileno(), 0x8915, struct.pack("256s", ifname[:15])  # SIOCGIFADDR
         )[20:24]
     )
+
+
+def get_closest_net_devices(gpu_dev):
```
Comment (quasiben) [inline ucp/utils.py]: This is not related to this PR but do you think we could/should start using this over `UCXPY_IFNAME` ?

File: setup.py [hunk 2]
```diff
@@ -12,7 +12,9 @@
 from setuptools import setup
 from setuptools.extension import Extension
 
-include_dirs = [os.path.dirname(get_python_inc())]
+include_dirs = [os.path.dirname(get_python_inc())] + [
```
Comment (quasiben) [inline setup.py]: Is this leftover from debugging sessions ?

**Comment by quasiben** [comment]
This looks good to me -- left a few small comments.  I imagine this will be particularly useful for MPI style testing of ucx-py, correct ?

File: setup.py (hunk 2 above)
Comment (jakirkham) [reply setup.py]: Assuming you are meaning the line below. Yeah this should already be the same as `os.path.dirname(get_python_inc())` unless I'm missing something.

Comment (madsbk) [reply setup.py]: Yes, this was included by mistake :)

File: ucp/utils.py (hunk 1 above)
Comment (madsbk) [reply ucp/utils.py]: Yes, I was hoping we could begin using it. 
But it properly best if it is possible to overwrite it using `UCXPY_IFNAME`.

**Comment by madsbk** [comment]
rerun test

**Comment by madsbk** [comment]
rerun tests

File: ucp/utils.py (hunk 1 above)
Comment (quasiben) [reply ucp/utils.py]: Makes sense

**Pull request merged**